etcdadm/etcdadm

```diff
@@ -0,0 +1,1559 @@
+#!/bin/bash
+
+# To prevent etcdctl errors like "Error:  net/http: TLS handshake timeout" when ETCD_ENDPOINT is set in e.g. /etc/environment
+# Gotcha: ETCDCTL_ENDPOINT seems to be preferred over --peers flag, which results in `etcdctl --peers` to be useless
+unset ETCDCTL_ENDPOINT
+
+export | grep ETCDCTL
+
+ETCD_WORK_DIR=${ETCD_WORK_DIR:-$(pwd)/work}
+
+if [ "${DEBUG:-}" == "yes" ]; then
+  set -vx
+fi
+
+_info() {
+  echo "$0: info: ${FUNCNAME[1]}: $*" >&2
+}
+
+_error() {
+  echo "$0: error: ${FUNCNAME[1]}: $*" >&2
+}
+
+_panic() {
+  echo "$0: panic: ${FUNCNAME[1]}: $*" >&2
+  exit 1
+}
+
+_array_join() {
+  local r=""
+  for t in "${@}"; do
+    if [ "$r" != "" ]; then
+      r="$r \"$t\""
+    else
+      r="\"$t\""
+    fi
+  done
+  echo "$r"
+}
+
+_current_time() {
+  date +%s
+}
+
+_run_as_root() {
+  local status
+  if [[ "$EUID" > 0 ]]; then
+    echo running command as root: "$(_array_join "$@")"
+    _array_join "${@}" | sudo bash
+  else
+    _array_join "${@}" | bash
+  fi
+  status=$?
+  return "$status"
+}
+
+config_awscli_docker_image() {
+  echo "${ETCDADM_AWSCLI_DOCKER_IMAGE:-quay.io/coreos/awscli}"
+}
+
+awscli_rkt_image() {
+  echo "docker://$(config_awscli_docker_image)"
+}
+
+config_aws_region() {
+  echo "${AWS_DEFAULT_REGION:-ap-northeast-1}"
+}
+
+config_aws_access_key_id() {
+  echo "${AWS_ACCESS_KEY_ID}"
+}
+
+config_aws_secret_access_key() {
+  echo "${AWS_SECRET_ACCESS_KEY}"
+}
+
+config_etcd_initial_cluster() {
+  echo "${ETCD_INITIAL_CLUSTER}"
+}
+
+config_etcd_endpoints() {
+  echo "${ETCD_ENDPOINTS}"
+}
+
+config_etcd_version() {
+  echo "${ETCD_VERSION:-3.1.3}"
+}
+
+config_member_count() {
+  echo "${ETCDADM_MEMBER_COUNT:-3}"
+}
+
+config_member_index() {
+  echo "${ETCDADM_MEMBER_INDEX}"
+}
+
+config_member_systemd_unit_name() {
+  echo "${ETCDADM_MEMBER_SYSTEMD_UNIT_NAME:-$(config_member_systemd_service_name).service}"
+}
+
+config_member_systemd_service_name() {
+  echo "${ETCDADM_MEMBER_SYSTEMD_SERVICE_NAME:-etcd-member-$(config_member_index)}"
+}
+
+config_cluster_snapshots_s3_uri() {
+  echo "${ETCDADM_CLUSTER_SNAPSHOTS_S3_URI:?missing required env}"
+}
+
+config_state_dir() {
+  echo "${ETCDADM_STATE_FILES_DIR:-/var/run/coreos/$(member_name)-state}"
+}
+
+cluster_etcd_aci_url() {
+  local v
+  v=v$(config_etcd_version)
+  echo "https://github.com/coreos/etcd/releases/download/$v/etcd-$v-linux-amd64.aci"
+}
+
+cluster_member_indices() {
```

Comment:
you are right, `seq 0 $(( $(config_member_count) -1 ))` is good enough probably :)

alternative is `for x in {0..$(( $(config_member_count) -1 ))}`